cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -415,6 +407,26 @@ func containsVersion(versions *unversioned.APIVersions, version string) bool {
 	return false
 }
 
+func (s *CMServer) buildKubeconfig() (*client.Config, error) {
```

Comment:
This function is identical in all three files. Can we abstract it out (by passing in the kubeconfig & master args) and just implement it once? Then next time we want to tweak the behavior we won't have to do it in three places.